src/dex/algebra/algebra.ts

```diff
@@ -147,6 +157,35 @@ export class Algebra extends SimpleExchange implements IDex<AlgebraData> {
     }
   }
 
+  /*
+   * When a non existing pool is queried, it's blacklisted for an arbitrary long period in order to prevent issuing too many rpc calls
+   * Once the pool is created, it gets immediately flagged
+   */
+  onPoolCreatedDeleteFromNonExistingSet: OnPoolCreatedCallback = ({
+    token0,
+    token1,
+  }) => {
+    const logPrefix = '[Algebra.onPoolCreatedDeleteFromNonExistingSet]';
+    const [_token0, _token1] = this._sortTokens(token0, token1);
+    const poolKey = `${token0}_${token1}`.toLowerCase();
+
+    // consider doing it only from master pool for less calls to distant cache
+
+    // delete entry locally to let local instance discover the pool
+    delete this.eventPools[this.getPoolIdentifier(_token0, _token1)];
+
+    this.logger.info(
+      `${logPrefix} delete pool from not existing set: ${poolKey}`,
+    );
+    // delete pool record from set, not waiting for result
+    this.dexHelper.cache.zrem(this.notExistingPoolSetKey, [poolKey]);
```

Comment:
I feel like better to add `catch`, so if it fails, it won't fail in arbitrary place, but will be contained